### PR TITLE
Refactor browse command

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

- Define custom formater along with model definition.
- Refactor `browse` and `service-versions`to use those new methods.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by PaulFarault** [comment]
We should wait for #429 to be merged before this one.

**Comment by PaulFarault** [comment]
This PR is now ready.

**Comment by mdrutel** [comment]
I think there's a problem with the timezone when displaying a datetime :
Date and time of the operation :
```
2023-09-11 12:36:20,972 - DEBUG - tdp.deployment_runner._run_operation - Running operation exporter_jmx_install
2023-09-11 12:36:20,972 - INFO - tdp.ansible_executor.execute - [DRY MODE] Ansible command: ansible-playbook /home/mdrutel01/projets/git/tdp-ops/tdp/ansible/ansible_collections/tosit/tdp/playbooks/exporter_jmx_install.yml
```
Date an time displayed by ```tdp browse <id>``` :
```
Deployment details
id               2
options          {'restart': 'False'}
start_time       2023-09-11 10:36:20
end_time         2023-09-11 10:36:28
status           SUCCESS
deployment_type  DAG

Operations
  deployment_id    operation_order  operation                                 host    extra_vars    start_time           end_time             state    logs
---------------  -----------------  ----------------------------------------  ------  ------------  -------------------  -------------------  -------  ------
              2                  1  exporter_jmx_install                      None    None          2023-09-11 10:36:20  2023-09-11 10:36:20  SUCCESS

```
Technically, it's correct (we are at UTC+2), but I prefer no to convert to the right timezone ^^'

File: tdp/core/models/base.py [hunk 1]
```diff
@@ -1,23 +1,60 @@
...
+                {key: self._formater(key, value) for key, value in value.items()}
+            )
+        elif isinstance(value, list):
+            if len(value) > 2:
+                return f"[{value[0]}, ..., {value[-1]}]"
+            return str(value)
+        elif isinstance(value, BaseEnum):
+            return value.name
+        elif isinstance(value, datetime):
+            return value.astimezone(LOCAL_TIMEZONE).strftime("%Y-%m-%d %H:%M:%S")
```
Comment (mdrutel) [inline tdp/core/models/base.py:59]: I think that this line may be written like :
```
return value.replace(tzinfo=timezone.utc).astimezone(LOCAL_TIMEZONE).strftime("%Y-%m-%d %H:%M:%S")
```
(I copied and adapted the line from the old browse.py)

Comment (PaulFarault) [reply tdp/core/models/base.py:59]: I'll do it to not modify the logic but I think that we could do better when dealing with timezones later.

Comment (PaulFarault) [reply tdp/core/models/base.py]: Done

**Review by mdrutel — APPROVED** [review]
tested OK ^^

**Review by sergkudinov — APPROVED** [review]
lgtm

**Pull request merged**